src/components/Payroll/PayrollLanding/PayrollLanding.test.tsx

```diff
@@ -0,0 +1,190 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { PayrollLanding } from './PayrollLanding'
+import { renderWithProviders } from '@/test-utils/renderWithProviders'
+
+// Mock the child components
+vi.mock('../RunPayrollFlow/RunPayroll', () => ({
+  RunPayroll: ({
+    companyId,
+    onEvent,
+  }: {
+    companyId: string
+    onEvent: (event: string, payload: unknown) => void
+  }) => (
+    <div data-testid="run-payroll">
+      <h2>Run Payroll</h2>
+      <p>Company ID: {companyId}</p>
+      <button
+        onClick={() => {
+          onEvent('test-event', {})
+        }}
+      >
+        Test Event
+      </button>
+    </div>
+  ),
+}))
+
+vi.mock('../PayrollHistory/PayrollHistory', () => ({
+  PayrollHistory: ({
+    companyId,
+    onEvent,
+  }: {
+    companyId: string
+    onEvent: (event: string, payload: unknown) => void
+  }) => (
+    <div data-testid="payroll-history">
+      <h2>Payroll History</h2>
+      <p>Company ID: {companyId}</p>
+      <button
+        onClick={() => {
+          onEvent('test-event', {})
+        }}
+      >
+        Test Event
+      </button>
+    </div>
+  ),
+}))
```

Comment:
Is it preferable to mock these? i wonder if it would be more aligned with our testing approach to configure the test fixtures so they could get mocked with msw